consensus/src/sync/light/validity_window.rs

```diff
@@ -0,0 +1,304 @@
+        verifier_block_number: u32,
+        expected_root: Blake2bHash,
+        validity_window_start: u32,
+        election_in_window: bool,
+    ) {
+        if self.validity_requests.is_none() {
+            // In this case we are going to start a new request
+            let epoch_number = Policy::epoch_at(verifier_block_number);
+            let chunk_index = 0;
+
+            self.validity_requests = Some(ValidityChunkRequest {
+                verifier_block_number,
+                root_hash: expected_root,
+                chunk_index,
+                validity_start: validity_window_start,
+                election_in_window,
+            });
+
+            // Add the peer
+            self.validity_queue.add_peer(peer_id);
+            self.syncing_peers.insert(peer_id);
+
+            let request = RequestHistoryChunk {
+                epoch_number,
+                block_number: verifier_block_number,
+                chunk_index: chunk_index as u64,
+            };
+
+            log::debug!(
+                verifier_bn = verifier_block_number,
+                chunk_index = chunk_index,
+                epoch = epoch_number,
+                validity_start = validity_window_start,
+                election_in_between = election_in_window,
+                "Starting validity window synchronization process"
+            );
+
+            // If we already have some history, we need to do a cleanup before starting the sync process
+            match &self.blockchain {
+                BlockchainProxy::Full(blockchain) => {
+                    let mut blockchain_wr = blockchain.write();
+                    if blockchain_wr
+                        .history_store
+                        .length_at(verifier_block_number, None)
+                        > 0
+                    {
+                        blockchain_wr.remove_epoch_history(epoch_number);
+
+                        log::debug!(epoch = epoch_number, "Removing existing history from epoch");
+                    }
+                }
+                BlockchainProxy::Light(_) => {
+                    unreachable!()
+                }
+            };
+
+            // Request the chunk
+            self.validity_queue.add_ids(vec![(request, None)]);
+        } else {
+            // If we are already requesting chunks, then we only add the peer
+            self.validity_queue.add_peer(peer_id);
+            self.syncing_peers.insert(peer_id);
+        }
+    }
+
+    pub fn start_validity_synchronization(&mut self, peer_id: TNetwork::PeerId) {
+        let macro_head = self.blockchain.read().macro_head();
+
+        let validity_start = macro_head
+            .block_number()
+            .saturating_sub(Policy::transaction_validity_window_blocks());
+
+        let validity_window_bn = if validity_start <= Policy::genesis_block_number() {
+            Policy::genesis_block_number()
+        } else {
+            Policy::election_block_before(validity_start)
```

Comment:
What if `validity_start` is already an election macro block?

Reply:
we need to sync the previous epoch in order to verify the history